include/mbgl/actor/actor_ref.hpp

```diff
@@ -24,19 +24,19 @@ template <class Object>
 class ActorRef {
 public:
     ActorRef(Object& object_, std::weak_ptr<Mailbox> weakMailbox_)
-        : object(object_),
+        : object(&object_),
           weakMailbox(std::move(weakMailbox_)) {
     }
 
     template <typename Fn, class... Args>
     void invoke(Fn fn, Args&&... args) {
         if (auto mailbox = weakMailbox.lock()) {
-            mailbox->push(actor::makeMessage(object, fn, std::forward<Args>(args)...));
+            mailbox->push(actor::makeMessage(*object, fn, std::forward<Args>(args)...));
         }
     }
 
 private:
-    Object& object;
+    Object* object;
```

Comment:
Using references on attributes implicitly deletes the default constructor, so I would not be able to do:

```
void OnlineFileSource::setResourceTransform(optional<ActorRef<ResourceTransform>>&& transform) {
    resourceTransform = std::move(transform);
}
```